sklearn/multioutput.py

```diff
@@ -700,7 +718,7 @@ def fit(self, X, Y):
         ]
         return self
 
-    @if_delegate_has_method("base_estimator")
+    @_BaseChain._available_if_base_estimator_has("predict_proba")
```

Comment:
I think this is a weird limitation of `available_if`. Subclasses require the parent's class to correctly resolve `_available_if_base_estimator_has `. This was the primary motivation of my suggestion at: https://github.com/scikit-learn/scikit-learn/pull/19948#discussion_r661514314


Maybe with @adrinjalali idea of having a `base_estimator` and #19948 where we pass in `self.attribute_name`:

```python
def _delegate_if_all_has_method(base_estimator):
    def _check(self, attribute_name):
        return hasattr(getattr(self, base_estimator), attribute_name) or all(
            hasattr(est, attribute_name) for est in self.estimators_
        )
    return available_if(_check)

class _MultiOutputEstimator(...):
    @_delegate_if_all_has_method("estimator")
    def partial_fit(self, X, y, classes=None, sample_weight=None):
        ...

class ClassifierChain(...):
    @ _delegate_if_all_has_method("base_estimator")
    def predict_proba(self, X):
        ...
```

Reply:
Sorry, but I don't find your solutions with a lambda or in this comment easy to read, or significantly easier to develop.

> Subclasses require the parent's class to correctly resolve _available_if_base_estimator_has . 

This is not a generic thing about subclasses, but a weirdness about the inconsistent mix of `base_estimator` and `estimator` within one file, which makes it harder to intelligibly use a module-level function. Should we rename them, rather, and use a single function?

Reply:
would you rename them to `estimator`? I think that'd make sense.

Reply:
I think you're missing the word `base_estimator` or similar in `_available_if_has_method_multioutput`. I don't think your naming is readable.
I've pushed another variant. I hope it works for you.